### ISSUE TITLE
Clarify Muon momentum convention: SGD-style momentum vs EMA-style first moment

### ISSUE THREAD

**Issue by MaeChd** [NVIDIA-NeMo/Emerging-Optimizers#166]
`OrthogonalizedOptimizer.step` currently implements the momentum candidate as an EMA-style first moment:

```python
M_t = beta * M_{t-1} + (1 - beta) * G_t
C_t = beta * M_t + (1 - beta) * G_t   # when nesterov=True
```

This differs from the SGD / PyTorch-SGD-style momentum convention, where I would expect:

```python
M_t = beta * M_{t-1} + G_t
C_t = G_t + beta * M_t                # when nesterov=True
```

The naming and docs currently make this convention ambiguous: `momentum` and `nesterov` are described as parameters of the internal SGD, and the Muon docs say that Muon runs “standard SGD-momentum with Nesterov momentum”. Keller Jordan’s Muon writeup also describes Muon as orthogonalizing the update generated by SGD momentum / SGD-Nesterov momentum.

At the same time, I understand why the current EMA-style form may have been chosen. Several recent Muon theory papers use this convention, for example:

- `M_t = beta M_{t-1} + (1 - beta) grad_t`
- `C_t = beta M_t + (1 - beta) grad_t` for Nesterov-style Muon

For exact polar decomposition, or a Newton-Schulz implementation that first normalizes the input, the two conventions are mostly scale-equivalent if the momentum buffer starts from zero:

```text
M_t^EMA = (1 - beta) * M_t^SGD
C_t^EMA = (1 - beta) * C_t^SGD
```

Since `Ortho(cG) = Ortho(G)` for positive scalar `c`, the final Muon direction can be identical in the ideal scale-invariant path.

The concern is that this constant factor can still matter for API semantics and comparisons:

- the optimizer state magnitude differs by a constant factor;
- checkpoint/state compatibility across implementations becomes ambiguous;
- subclasses of `OrthogonalizedOptimizer` may not be scale-invariant;
- numerical epsilon / finite precision paths may not be exactly scale-invariant;
- learning-rate or momentum comparisons against SGD-style Muon implementations can be confusing.

### Question

Is the EMA-style momentum update intentionally the canonical convention for this repository, or should `OrthogonalizedOptimizer` use SGD-style momentum to match the “standard SGD-momentum” wording?

If EMA-style is intentional, would you be open to a documentation/comment clarification, e.g. saying “EMA-style first moment used before orthogonalization” rather than “standard SGD-momentum”?

If SGD-style semantics are intended, I can send a PR changing the update and adding a small unit test.
```

References：
- Keller Jordan Muon writeup: https://kellerjordan.github.io/posts/muon/
- Emerging-Optimizers Muon docs: https://docs.nvidia.com/nemo/emerging-optimizers/latest/_modules/emerging_optimizers/orthogonalized_optimizers/muon.html
- Emerging-Optimizers current momentum code: https://docs.nvidia.com/nemo/emerging-optimizers/latest/_modules/emerging_optimizers/orthogonalized_optimizers/orthogonalized_optimizer.html
- EMA-style theory analysis: https://arxiv.org/pdf/2507.01598, https://arxiv.org/pdf/2509.15816, https://arxiv.org/pdf/2601.19400

**Comment by skyw** [comment]
Right, terms like momentum and weight decay are heavily overloaded and are mostly beyond repair.

The EMA choice was intentional because most implementation preceding this use EMA, @mkhona-nvidia should chime in.

Document should be updated to be more explicit. Code change is not an option because it would break so many things. Adding a flag to select algorithm is technically possible, but also need to be very careful.

**Comment by skyw** [comment]
Also FWIW, this is pytorch does https://github.com/pytorch/pytorch/blob/4de309fa4a0f37189cd69363f1559b9f68b57d35/torch/optim/_muon.py#L333

**Comment by mkhona-nvidia** [comment]
Hi @MaeChd, what type of momentum mechanism to use is a controversial topic. However, we think it does not fundamentally change the quality of the final model, but rather affects what hyperparameters are optimal. As you've pointed out, since the output of newton-schulz (and generally, polar decomposition) is scale invariant, we do not think it would affect results too much. Choices are all over the place: Scion and it's derivatives (see https://arxiv.org/abs/2502.07529 for example) use EMA-style momentum while Kimi-Muon and it's derivatives use "theory style" (https://arxiv.org/abs/2502.16982) and also the nanoGPT speedrun. 

@skyw, we should add in the documentation that we use EMA-style momentum as our default. We might also think about a "momentum mixin", the same way we have a "WeightDecayMixin" which defaults to EMA-style momentum. This would be for users who, to give one example, might take a pretrained checkpoint that used theory-style momentum and want to continue using theory-style momentum for hyperparameter reusage.

**Comment by MaeChd** [comment]
@skyw @mkhona-nvidia , thanks for the clarification, that makes sense. Keeping EMA-style momentum as the default for compatibility sounds reasonable.

A docs clarification would probably resolve the ambiguity here. It would be helpful to spell out the exact convention used by the implementation, for example:

```text
B_t = beta * B_{t-1} + (1 - beta) * G_t
C_t = beta * B_t + (1 - beta) * G_t    # if nesterov=True
C_t = B_t                              # otherwise
```

and maybe add a short note that this is scale-equivalent to the unscaled / theory-style convention under scale-invariant orthogonalization, but changes the momentum-buffer scale and can matter for hyperparameter reuse, checkpoint continuation, or code paths that expose or consume the momentum state.

@skyw , the PyTorch Muon example you linked is actually a good illustration of why this is worth spelling out: the implementation uses EMA-style momentum, while the rendered algorithm description uses the unscaled buffer convention.
